src/gui/torrentcontentmodel.cpp

```diff
@@ -88,6 +96,40 @@ namespace
             return QIcon(iconPixmap);
         }
     };
+#elif defined(Q_OS_MAC)
+    // There is a similar bug on macOS, to be reported to Qt
+    // https://github.com/qbittorrent/qBittorrent/pull/6156#issuecomment-316302615
+    class MacFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            QSize sz(32, 32);
+            QPixmap pixmap = pixmapForExtension(info.suffix(), sz);
```

Comment:
I would not pass empty suffixes at all (would add a check right here). Two related questions:
1. Do double extension (.tar.gz as an example) require special handling?
2. What to do with a directories whose names contain an extension?